dpr_reader.py

```diff
@@ -105,26 +107,33 @@ def rank(
         if not docs:
             return None
 
-        traversal_paths = parameters.get('traversal_paths', self.traversal_paths)
+        traversal_paths = parameters.get(
+            'traversal_paths',
+            self.traversal_paths
+        )
         batch_size = parameters.get('batch_size', self.batch_size)
 
-        for doc in docs.traverse_flat(traversal_paths):
-            if not getattr(doc, 'text'):
-                continue
-
+        for doc in docs.traverse_flat(traversal_paths, filter_fn=lambda x: bool(x.text)):
             new_matches = []
 
             doc_arr = DocumentArray([doc])
-            # match_batches_generator = doc_arr.batch(
-            #     traversal_paths=['m'],
-            #     batch_size=batch_size,
-            #     require_attr='text',
-            # )
-            match_batches_generator = doc_arr.traverse_flat(traversal_paths=['m'], filter_fn= lambda x: bool(x)).batch(batch_size=batch_size)
+
+            match_batches_generator = doc_arr.traverse_flat(
+                traversal_paths=['m'],
+                filter_fn=lambda x: bool(x.text)
```

Comment:
Is this filter function identical to `require_attr='text'` ?

Reply:
Looks equivalent yes